package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn

```diff
@@ -7,13 +7,15 @@ local util = require 'gluon.util'
 local uci = require('simple-uci').cursor()
 
 local vpn_core = require 'gluon.mesh-vpn'
+local vpn, active_vpn = vpn_core.get_active_provider()
 
 uci:section('network', 'interface', 'mesh_vpn', {
 	ifname = vpn_core.get_interface(),
-	proto = 'gluon_mesh',
+	proto = active_vpn.proto(),
 	transitive = true,
 	fixed_mtu = true,
-	macaddr = util.generate_mac(7),
+	macaddr = util.generate_mac(7), -- needed for proto gluon_mesh
+	index = 7,                      -- needed for proto gluon_wireguard
```

Comment:
We wouldn't need to change anything in this UCI section if `mesh-vpn` is already the VXLAN interface.

Reply:
I think, this will actually be a smart solution.

I suspect, this will become the third rewrite of the gluon_wireguard.sh netifd proto. And tbh, developing netifd protos is kind of pain. -.-

Reply:
Done.